presto-common/src/main/java/com/facebook/presto/common/block/AbstractRowBlock.java

```diff
@@ -124,6 +126,36 @@ public Block getRegion(int position, int length)
         return createRowBlockInternal(position + getOffsetBase(), length, getRowIsNull(), getFieldBlockOffsets(), getRawFieldBlocks());
     }
 
+    @Override
+    public final OptionalInt fixedSizeInBytesPerPosition()
+    {
+        OptionalInt fieldSize = fixedSizeInBytesPerFieldPosition();
```

Comment:
If `fixedSizeInBytesPerPosition` is called multiple times, `fixedSizeInBytesPerFieldPosition` would be called multiple times. Can we make `fieldSize` a field of the class and set it the first time `fixedSizeInBytesPerPosition` is called?

Reply:
That shouldn't be a problem, because all calls to `Block#fixedSizeInBytesPerPosition()` are initiated at their root by  a wrapping `DictionaryBlock` layer- which it caches. All implementations of `fixedSizeInBytes()` are also very cheap, with the most expensive ones being `RowBlock` (virtual method dispatch proportional to the number of field blocks) and `DictionaryBlock` (a volatile read, potentially followed by a virtual method dispatch to `dictionary.fixedSizeInBytesPerPosition()`).

Now, if in the future someone chose to use `Block#fixedSizeInBytesPerPosition()` to compute `RowBlock#getRegionSizeInBytes(position, length)` then this _might_ end up being called more frequently- but the current implementation doesn't do that- so I'm not sure it justifies adding a new field to store the result.